api/models.py

```diff
@@ -77,6 +78,16 @@ def convert_values_to_string(cls, elements):
                 converted_elements[key] = json.dumps(element)
         return converted_elements
 
+    @classmethod
+    def order_by_weight(cls, elements):
+        ordered_elements = elements.copy()
+        for name, element in elements.items():
+            if element.get('type') == 'fieldset':
+                ordered_elements[name]['elements'] = cls.order_by_weight(element['elements'])
+        return OrderedDict(
+            sorted(ordered_elements.items(), key=lambda element: element[1].get('weight', 10000))
```

Comment:
Instead of a static 10k, we also could use ``sys.maxint``. @keimlink, what do you think about this?